hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieROTablePathFilter.java

```diff
@@ -56,6 +58,15 @@
 public class HoodieROTablePathFilter implements Configurable, PathFilter, Serializable {
 
   private static final long serialVersionUID = 1L;
+  public static final String METADATA_PREFIX = "hoodie.metadata";
+
+  // List using internal metadata table which saves file listings
+  public static final String METADATA_ENABLE_PROP = METADATA_PREFIX + ".enable";
+  public static final boolean DEFAULT_METADATA_ENABLE = false;
```

Comment:
This default can be true as within HoodieTableMetadata - If the metadata table does not exist, RPC calls are used to retrieve file listings from the file system.

Reply:
Makes sense. I introduced a `true` default for the readers, but just a but concerned about the additional overhead this introduces in the regular (non metadata table) hudi path, as it would keep checking for the metadata folder and call `getFileStatus()` and then fail. This can be an overhead for S3.